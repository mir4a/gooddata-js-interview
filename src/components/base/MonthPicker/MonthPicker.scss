.MonthPicker {
  display: inline;

  & &__select {
    font-size: 16px;
    
    h1 & {
      font-size: inherit;
      margin: 0 10px;
    }
  }
}
