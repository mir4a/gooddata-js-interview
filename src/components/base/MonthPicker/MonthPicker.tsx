import React, { ChangeEvent, useCallback } from 'react'
import './MonthPicker.scss'

interface MonthPickerProps {
  onChange: Function;
  defaultValue?: number;
}

// eslint-disable-next-line max-len
export default function MonthPicker(props: MonthPickerProps): React.ReactElement {
  const { defaultValue = '1', onChange } = props;

  const handleOnChange = useCallback(
    (event: ChangeEvent<HTMLSelectElement>) => {
    console.log('onchange')
    onChange && onChange(event)
  }, [onChange])

  return (
    <div className="MonthPicker">
      <select
        defaultValue={defaultValue}
        onChange={handleOnChange}
        className="MonthPicker--select"
      >
        <option value="1">January</option>
        <option value="2">February</option>
        <option value="3">March</option>
        <option value="4">April</option>
        <option value="5">May</option>
        <option value="6">June</option>
        <option value="7">July</option>
        <option value="8">August</option>
        <option value="9">September</option>
        <option value="10">October</option>
        <option value="11">November</option>
        <option value="12">December</option>
      </select>
    </div>
  );
}