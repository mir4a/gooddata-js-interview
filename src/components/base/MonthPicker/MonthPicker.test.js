import React from 'react';
import ReactDOM from 'react-dom';
import MonthPicker from '.';
import renderer from 'react-test-renderer';

it('renders correct', async () => {
  const div = document.createElement('div');
  const onChange = jest.fn();
  ReactDOM.render(<MonthPicker onChange={onChange} />, div);
});

it('match basic snapshot', () => {
  const onChange = jest.fn();
  const tree = renderer.create(<MonthPicker onChange={onChange} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('render with defaultValue', () => {
  const onChange = jest.fn();
  const tree = renderer
    .create(<MonthPicker onChange={onChange} defaultValue="2" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
