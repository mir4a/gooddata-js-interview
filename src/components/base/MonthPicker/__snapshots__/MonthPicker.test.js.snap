// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match basic snapshot 1`] = `
<div
  className="MonthPicker"
>
  <select
    className="MonthPicker--select"
    defaultValue="1"
    onChange={[Function]}
  >
    <option
      value="1"
    >
      January
    </option>
    <option
      value="2"
    >
      February
    </option>
    <option
      value="3"
    >
      March
    </option>
    <option
      value="4"
    >
      April
    </option>
    <option
      value="5"
    >
      May
    </option>
    <option
      value="6"
    >
      June
    </option>
    <option
      value="7"
    >
      July
    </option>
    <option
      value="8"
    >
      August
    </option>
    <option
      value="9"
    >
      September
    </option>
    <option
      value="10"
    >
      October
    </option>
    <option
      value="11"
    >
      November
    </option>
    <option
      value="12"
    >
      December
    </option>
  </select>
</div>
`;

exports[`render with defaultValue 1`] = `
<div
  className="MonthPicker"
>
  <select
    className="MonthPicker--select"
    defaultValue="2"
    onChange={[Function]}
  >
    <option
      value="1"
    >
      January
    </option>
    <option
      value="2"
    >
      February
    </option>
    <option
      value="3"
    >
      March
    </option>
    <option
      value="4"
    >
      April
    </option>
    <option
      value="5"
    >
      May
    </option>
    <option
      value="6"
    >
      June
    </option>
    <option
      value="7"
    >
      July
    </option>
    <option
      value="8"
    >
      August
    </option>
    <option
      value="9"
    >
      September
    </option>
    <option
      value="10"
    >
      October
    </option>
    <option
      value="11"
    >
      November
    </option>
    <option
      value="12"
    >
      December
    </option>
  </select>
</div>
`;
